prescient/simulator/prescient.py

```diff
@@ -11,127 +11,76 @@
 ####################################################
 
-import traceback
 import sys
-import os
-import pyutilib
-import profile
-from . import master_options as MasterOptions
+
+import prescient.plugins.internal
+import prescient.simulator.config
+
+from .config import PrescientConfig
 from .options import Options
 from .simulator import Simulator
 from .data_manager import DataManager
 from .time_manager import TimeManager
 from .oracle_manager import OracleManager
 from .stats_manager import StatsManager
 from .reporting_manager import ReportingManager
-from prescient.stats.overall_stats import OverallStats
-import prescient.plugins 
-
+from prescient.scripts import runner
 from prescient.engine.egret import EgretEngine as Engine
 
-try:
-    import pstats
-    pstats_available=True
-except ImportError:
-    pstats_available=False
-
-def create_prescient(options: Options):
-    engine = Engine()
-    time_manager = TimeManager()
-    data_manager = DataManager()
-    oracle_manager = OracleManager()
-    stats_manager = StatsManager()
-    reporting_manager = ReportingManager()
-    prescient = Simulator(engine, time_manager, data_manager, oracle_manager, stats_manager, reporting_manager)
-    return prescient
-
-
-def main_prescient(options: Options):
-    ans = None
-
-    if pstats_available and options.profile > 0:
-        #
-        # Call the main ef writer with profiling.
-        #
-        tfile = pyutilib.services.TempfileManager.create_tempfile(suffix=".profile")
-        tmp = profile.runctx('simulate(options)', globals(), locals(), tfile)
-        p = pstats.Stats(tfile).strip_dirs()
-        p.sort_stats('time', 'cumulative')
-        p = p.print_stats(options.profile)
-        p.print_callers(options.profile)
-        p.print_callees(options.profile)
-        p = p.sort_stats('cumulative', 'calls')
-        p.print_stats(options.profile)
-        p.print_callers(options.profile)
-        p.print_callees(options.profile)
-        p = p.sort_stats('calls')
-        p.print_stats(options.profile)
-        p.print_callers(options.profile)
-        p.print_callees(options.profile)
-        pyutilib.services.TempfileManager.clear_tempfiles()
-        ans = [tmp, None]
-
-    else:
-        if options.traceback is True:
-            simulator = create_prescient(options)
-            ans = simulator.simulate(options)
+class Prescient(Simulator):
+
+    def __init__(self):
+
+        self.config = PrescientConfig()
+
+        engine = Engine()
+        time_manager = TimeManager()
+        data_manager = DataManager()
+        oracle_manager = OracleManager()
+        stats_manager = StatsManager()
+        reporting_manager = ReportingManager()
+
+        self.simulate_called = False
+
+        super().__init__(engine, time_manager, data_manager, oracle_manager, 
+                         stats_manager, reporting_manager, 
+                         self.config.plugin_context.callback_manager)
+
+    def simulate(self, **options):
```

Comment:
Should the options be specified here or in `__init__`? Now that the plugin registration is per-instance, I think it makes better sense to be in `__init__` -- though I don't have strong feelings either way.